packages/editor/src/editor/sidebar/fields/BadgeField.tsx

```diff
@@ -0,0 +1,49 @@
+import { BasicField, InputBadge, InputBadgeArea, splitNewLine, type BasicFieldProps } from '@axonivy/ui-components';
+import { useEffect, useRef, useState, type FocusEvent } from 'react';
+import { badgeProps } from '../../../utils/badge-properties';
+
+export type BadgeFieldProps = BasicFieldProps & {
+  value: string;
+  childRef: React.RefObject<HTMLElement>;
+  area?: boolean;
+};
+
+export const BadgeField = ({ value, label, children, message, childRef, area, ...props }: BadgeFieldProps) => {
```

Comment:
don't understand why this is abstracted on the field and not on the input...
Seems a lot more complicated to me.